public/js/actions/sources.js

```diff
@@ -12,17 +12,35 @@ const { getSource, getSourceText } = require("../selectors");
 const constants = require("../constants");
 const Prefs = require("../prefs");
 
-// delay is in ms
-const FETCH_SOURCE_RESPONSE_DELAY = 200;
+/**
+ * Throttles source dispatching to reduce rendering churn.
+ */
+let newSources = [];
+let newSourceTimeout;
+function queueSourcesDispatch(dispatch) {
+  if(!newSourceTimeout) {
+    // Even just batching every 10ms works because we just want to
+    // group sources that come in at once.
+    newSourceTimeout = setTimeout(() => {
+      dispatch({ type: constants.ADD_SOURCES,
+                 sources: newSources });
+      newSources = [];
+      newSourceTimeout = null;
+    }, 10);
+  }
+}
```

Comment:
`debounce` seems like the wrong abstraction though, at least for what `debounce` commonly means. Usually `debounce` means to run a function a certain amount of time after the debounced function is called, and if it is called again it resets the timer. We don't want that: we want to schedule a dispatch 10ms later, no matter what. It's more of a "throttle". Also, debounced functions don't usually depend on external state like that. This is an explicit function to queue a dispatch that can't be transparently debounced.

There probably is an abstraction here, maybe one that we could even tailor to actions specifically. I'd like to do that later though :)